src/fields/u64/fp/wrapper.rs

```diff
@@ -1,101 +1,195 @@
-use core::ops::{Add, Mul, Neg, Sub};
+use core::{ops::{Add, Mul, Neg, Sub}, marker::PhantomData};
 
-use super::fiat;
+use super::fiat::{self, FpMontgomeryDomainFieldElement};
+use ark_ff::{BigInt, SqrtPrecomputation};
+use ark_std::{
+    hash::Hash,
+};
+use derivative::Derivative;
 
 #[derive(Copy, Clone)]
-pub struct Fp(fiat::FpMontgomeryDomainFieldElement);
+pub struct Fp(FpMontgomeryDomainFieldElement);
+    
+pub trait FpTrait: Send + Sync + 'static + Sized {
```

Comment:
I don't think we should have a trait here. We should leave the existing `Fp` API unchanged (so we can use it for the embedded-friendly, arkworks-less case) and only add Arkworks impls under the `arkworks` submodule. Those impls should be implemented directly for the `Fp` type, rather than using a generic approach.

Then, we can copy-paste the implementations from Fp to Fq, Fr. The arkworks traits are already a complicated mess, and adding more generics will make them even harder to reason about. On the other hand, copy-pasting code is easy, and relatively simple to maintain going forward (we are not going to be changing the field implementations much).